### PR TITLE
Add thesis field to transfer processing form

### PR DESCRIPTION
**Pull request by matt-bernhardt**
This completes building the transfer processing form UI, adding the table of relevant theses. Future work will complete the actual work requested by the form.

#### Relevant ticket(s):

* https://mitlibraries.atlassian.net/browse/ETD-279
* https://mitlibraries.atlassian.net/browse/ETD-280

#### Please note:

I'm curious if there is a better way to peform the filtering. In testing, it appears that thesis records which carry multiple departments are appearing in the filtered list, but when rendered these theses would only show the filtered department name. Reloading the record would restore both, but I'm somewhat concerned that the way the filtering is being done too aggressively (i.e. the record is being manipulated, and if saved we would lose some thesis-department links)

#### Developer

- [x] All new ENV is documented in README
- [x] All new ENV has been added to Heroku Pipeline, Staging and Prod
- [x] ANDI or Wave has been run in accordance to
      [our guide](https://mitlibraries.github.io/guides/basics/a11y.html) and
      all issues introduced by these changes have been resolved or opened as new
      issues (link to those issues in the Pull Request details above)
- [ ] Stakeholder approval has been confirmed (or is not needed)

#### Code Reviewer

- [x] The commit message is clear and follows our guidelines
      (not just this pull request message)
- [x] There are appropriate tests covering any new functionality
- [x] The documentation has been updated or is unnecessary
- [x] The changes have been verified
- [x] New dependencies are appropriate or there were no changes

#### Requires database migrations?

NO

#### Includes new or updated dependencies?

NO

### CODE REVIEW

File: app/controllers/transfer_controller.rb [hunk 1]
```diff
@@ -42,7 +44,14 @@ def select
...
 
   def show
+    # Load the details of the requested Transfer record
     @transfer = Transfer.find(params[:id])
+
+    # Load the Thesis records for the period covered by this Transfer (the
+    # graduation month/year, and the department)
+    @theses = Thesis.all
+    @theses = @theses.where('grad_date == ?', @transfer.grad_date)
+    @theses = @theses.includes(:departments).where("departments.name_dw == ?", @transfer.department.name_dw).references(:departments)
```
Comment (JPrevost) [inline app/controllers/transfer_controller.rb:54]: When reading this it feels like it'll be a super inefficient set of queries, but ActiveRecord turns this into a single query.

To relieve that cognitive confusion as to what is going on, it may be clearer to change it slightly while not changing the actual query ActiveRecord will compile to under the hood like:

```
@theses = Thesis.where('grad_date == ?', @transfer.grad_date)
@theses = @theses.includes(:departments).where("departments.name_dw == ?", @transfer.department.name_dw).references(:departments)
```

If that doesn't read clearer to you that we aren't _actually_ loading all the Thesis then no worries on changing it as I confirmed it makes absolutely no difference on the compiled db query.

TLDR: ActiveRecord is smarter than I give it credit for and my brain doesn't trust it so I'd like to be nice to my future brain but this is absolutely not a required change.

File: app/views/transfer/_thesis_target.html.erb [hunk 2]
```diff
@@ -0,0 +1,16 @@
+<tr>
```
Comment (JPrevost) [inline app/views/transfer/_thesis_target.html.erb:1]: I think we'll want the author kerb displayed here. The file names are _supposed_ to include a kerb and that will be a really easy way for a processor to rapidly identify which files are most likely what they are looking for. If we need to remove something to fit it, I suspect at this stage the kerb is more important than degree (I would think that is more of a thesis processing step), but Jess could confirm whether my suspicious is correct.

**Review by JPrevost — APPROVED** [review]
I have two very much non-blocking comments. If you feel either are useful, feel free to change now or later. If you don't think they make useful changes, feel free to ignore them entirely :) 

This work is going in a very good direction.

File: app/controllers/transfer_controller.rb (hunk 1 above)
Comment (matt-bernhardt) [reply app/controllers/transfer_controller.rb:54]: Honestly, I wasn't sure whether this was going to be inefficient, or if a two-line approach (or even just chaining them all together into one expression) would be better. I'm happy to go with a two-line approach here - will push an updated commit in a bit.

File: app/views/transfer/_thesis_target.html.erb (hunk 2 above)
Comment (matt-bernhardt) [reply app/views/transfer/_thesis_target.html.erb:1]: I've pushed an update to this, adding the user's kerberos to this line - so it is new `kerbid: display_name`

**Review by JPrevost — APPROVED** [review]
